File: fixtures/phase5-multilang/go/vulnerable.go
package phase5

import (
    "net/http"
    "os"
    "os/exec"
    "github.com/gin-gonic/gin"
)

func routes(router *gin.Engine) {
    router.GET("/run", runCommand)
}

func runCommand(context *gin.Context) {
    input := context.Query("command")
    exec.Command("sh", "-c", input).Run()
    db.Query(input)
    os.ReadFile(input)
    http.Get(input)
    http.Redirect(context.Writer, context.Request, input, http.StatusFound)
}

File: fixtures/phase5-multilang/rust/src/vulnerable.rs
use axum::{Router, routing::get};
use std::process::Command;

fn routes() -> Router {
    Router::new().route("/run", get(run_command))
}

async fn run_command(input: String) {
    Command::new("sh").arg("-c").arg(input).output();
    sqlx::query(input);
    std::fs::read(input);
    reqwest::get(input).await;
    axum::response::Redirect::to(input);
}
